core/cont/inc/TList.h

```diff
@@ -70,7 +70,12 @@ friend  class TListIter;
    typedef TListIter Iterator_t;
 
    TList() : fAscending(kTRUE) { }
-   TList(TObject *) : fAscending(kTRUE) { } // for backward compatibility, don't use
+
+   R__DEPRECATED(6, 34,
+                 "Confusing, not explicit, looks like a copy ctor, but just discards the argument and default "
+                 "constructs instead, marked as don't use for two decades already")
+   TList(TObject *) : fAscending(kTRUE) {} // for backward compatibility, don't use
```

Comment:
```suggestion
   TList(TObject *) R__DEPRECATED(6, 34, "The argument is ignored. Use the default constructor TList::TList().") : fAscending(kTRUE) { } // for backward compatibility, don't use
```
The macro is intended for postfix usage. Also, we try to keep the warning as short as possible, focusing on the suggested alternative. If you want to give more context, you can do this in the release notes.

With my suggestion, the warnings would look like this:
```
repro.C: In function ‘void repro()’:
repro.C:4:29: warning: ‘TList::TList(TObject*)’ is deprecated: will be removed in ROOT v6.34: The argument is ignored. Use the default constructor TList::TList(). [-Wdeprecated-declarations]
    4 |    TList mylist = new TList();
      |                             ^
In file included from repro.C:1:
/home/rembserj/spaces/master/install/include/root/TList.h:73:4: note: declared here
   73 |    TList(TObject *) R__DEPRECATED(6, 34, "The argument is ignored. Use the default constructor TList::TList().") : fAscending(kTRUE) { } // for backward compatibility, don't use
      |  
```

Reply:
well, looks like you already have the commit written yourself, feel free to use your own. otherwise, this is a bit inefficient ;) if you like my other commit, you are welcome to cherry-pick it, otherwise, I don't mind it solved differently.